c_glib/arrow-glib/array-builder.h

```diff
@@ -782,6 +782,108 @@ gboolean garrow_string_array_builder_append(GArrowStringArrayBuilder *builder,
                                             GError **error);
 
 
+#define GARROW_TYPE_DATE32_ARRAY_BUILDER        \
+  (garrow_date32_array_builder_get_type())
+#define GARROW_DATE32_ARRAY_BUILDER(obj)                        \
+  (G_TYPE_CHECK_INSTANCE_CAST((obj),                            \
+                              GARROW_TYPE_DATE32_ARRAY_BUILDER, \
+                              GArrowDate32ArrayBuilder))
+#define GARROW_DATE32_ARRAY_BUILDER_CLASS(klass)                \
+  (G_TYPE_CHECK_CLASS_CAST((klass),                             \
+                           GARROW_TYPE_DATE32_ARRAY_BUILDER,    \
+                           GArrowDate32ArrayBuilderClass))
+#define GARROW_IS_DATE32_ARRAY_BUILDER(obj)                             \
+  (G_TYPE_CHECK_INSTANCE_TYPE((obj),                                    \
+                              GARROW_TYPE_DATE32_ARRAY_BUILDER))
+#define GARROW_IS_DATE32_ARRAY_BUILDER_CLASS(klass)             \
+  (G_TYPE_CHECK_CLASS_TYPE((klass),                             \
+                           GARROW_TYPE_DATE32_ARRAY_BUILDER))
+#define GARROW_DATE32_ARRAY_BUILDER_GET_CLASS(obj)              \
+  (G_TYPE_INSTANCE_GET_CLASS((obj),                             \
+                             GARROW_TYPE_DATE32_ARRAY_BUILDER,  \
+                             GArrowDate32ArrayBuilderClass))
```

Comment:
I wonder if having a code generation script would assist with automating the generation of these boilerplate macros. What do you think?

Reply:
It's reasonable.
I'll talk about with GLib developers. If they also think it's useful, we should share it.

Reply:
I heard at gtk-list mailing list and a developer tells me a convenient API: https://mail.gnome.org/archives/gtk-list/2017-September/msg00031.html

I'll migrate to the API later.